### ISSUE TITLE
Importing Slack export failed on user import

### ISSUE THREAD

**Issue by whren** [RocketChat/Rocket.Chat#4495]
Your Rocket.Chat version: 0.42.0-develop
Client on FireFox (because of #3050)

Importing Slack export failed with the following stack exception :

> Exception in defer callback: Error: Match error: Failed Match.OneOf, Match.Maybe or Match.Optional validation
    at exports.check (packages/check/match.js:34:1)
    at [object Object].Meteor.methods.setAvatarFromService (server/methods/setAvatarFromService.coffee:5:3)
    at [object Object].methodsMap.(anonymous function) (packages/rocketchat:lib/server/lib/debug.js:17:26)
    at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1711:12)
    at packages/ddp-server/livedata_server.js:1624:18
    at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
    at Server.apply (packages/ddp-server/livedata_server.js:1623:45)
    at Server.call (packages/ddp-server/livedata_server.js:1566:17)
    at [object Object].<anonymous> (packages/rocketchat_importer-slack/server.coffee:135:16)
    at packages/dispatch_run-as-user/packages/dispatch_run-as-user.js:186:1
    at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
    at Object.Meteor.runAsUser (packages/dispatch_run-as-user/packages/dispatch_run-as-user.js:185:1)
    at [object Object].<anonymous> (packages/rocketchat_importer-slack/server.coffee:127:15)
    at packages/dispatch_run-as-user/packages/dispatch_run-as-user.js:186:1
    at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
    at Object.Meteor.runAsUser (packages/dispatch_run-as-user/packages/dispatch_run-as-user.js:185:1)
    at packages/rocketchat_importer-slack/server.coffee:113:13
    at packages/rocketchat_importer-slack/server.coffee:112:5
    at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
    at packages/meteor/timers.js:6:1
    at runWithEnvironment (packages/meteor/dynamics_nodejs.js:110:1)

Once the error has been raised there is no more progress :

![image](https://cloud.githubusercontent.com/assets/1455341/18986741/86ab3544-86fe-11e6-8636-a2162954eb39.png)

Returning to the import menu show the following error (that's only a consequence) :

![image](https://cloud.githubusercontent.com/assets/1455341/18986730/7214188a-86fe-11e6-88ca-0876c5b30a44.png)

In fact 1 user is semi-created (not all information are filled).
The only way to recover from this is to restart the server.
Each time I retry the import it fails on the next user import for the same reason.

Debugging leads me to https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-importer-slack/server.coffee#L135 where the call to setAvatarFromService with null contextType argument fails in the exports.check

**Comment by lc3t35** [comment]
+1 same message, Rocket.Chat version: 0.42.0
````
setAvatarFromService -> userId: TnzZPnkY3ZboGiBXu , arguments:  { '0': 'https://avatars.slack-edge.com/2016-08-16/69851868434_db10940b218e032048b4_original.jpg',
rocketchat_1          |   '1': null,
rocketchat_1          |   '2': 'url' }
rocketchat_1          | Exception in defer callback: Error: Match error: Failed Match.OneOf, Match.Maybe or Match.Optional validation
rocketchat_1          |     at exports.check (packages/check/match.js:34:1)
rocketchat_1          |     at [object Object].Meteor.methods.setAvatarFromService (server/methods/setAvatarFromService.coffee:5:3)
rocketchat_1          |     at [object Object].methodsMap.(anonymous function) (packages/rocketchat:lib/server/lib/debug.js:17:26)
rocketchat_1          |     at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1711:12)
rocketchat_1          |     at packages/ddp-server/livedata_server.js:1624:18
rocketchat_1          |     at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
rocketchat_1          |     at Server.apply (packages/ddp-server/livedata_server.js:1623:45)
rocketchat_1          |     at Server.call (packages/ddp-server/livedata_server.js:1566:17)
rocketchat_1          |     at [object Object].<anonymous> (packages/rocketchat_importer-slack/server.coffee:135:16)
rocketchat_1          |     at packages/dispatch_run-as-user/packages/dispatch_run-as-user.js:186:1
rocketchat_1          |     at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
rocketchat_1          |     at Object.Meteor.runAsUser (packages/dispatch_run-as-user/packages/dispatch_run-as-user.js:185:1)
rocketchat_1          |     at [object Object].<anonymous> (packages/rocketchat_importer-slack/server.coffee:127:15)
rocketchat_1          |     at packages/dispatch_run-as-user/packages/dispatch_run-as-user.js:186:1
rocketchat_1          |     at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
rocketchat_1          |     at Object.Meteor.runAsUser (packages/dispatch_run-as-user/packages/dispatch_run-as-user.js:185:1)
rocketchat_1          |     at packages/rocketchat_importer-slack/server.coffee:113:13
rocketchat_1          |     at packages/rocketchat_importer-slack/server.coffee:112:5
rocketchat_1          |     at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
rocketchat_1          |     at packages/meteor/timers.js:6:1
rocketchat_1          |     at runWithEnvironment (packages/meteor/dynamics_nodejs.js:110:1)
````

**Comment by lc3t35** [comment]
I suggest to change null with empty string
````
Meteor.call 'setAvatarFromService', url, null, 'url'
````
by
````
Meteor.call 'setAvatarFromService', url, '', 'url'
````

**Comment by engelgabriel** [comment]
closed this in #4644

**Comment by engelgabriel** [comment]
The PR https://github.com/RocketChat/Rocket.Chat/pull/4604 tries to fix this in a different way. What is best?
